src/core/load-fragment.c

```diff
@@ -776,8 +776,94 @@ int config_parse_socket_bindtodevice(
+DEFINE_CONFIG_PARSE_ENUM(config_parse_output, exec_output, ExecOutput, "Failed to parse output literal specifier");
+
+int config_parse_exec_input(const char *unit,
+                            const char *filename,
+                            unsigned line,
+                            const char *section,
+                            unsigned section_line,
+                            const char *lvalue,
+                            int ltype,
+                            const char *rvalue,
+                            void *data,
+                            void *userdata) {
+        ExecContext *c = data;
+        const char *name;
+
+        assert(data);
+        assert(filename);
+        assert(line);
+        assert(rvalue);
+
+        name = startswith(rvalue, "fd:");
+        if (name) {
+                /* Strip prefix and validate fd name */
+                if (!fdname_is_valid(name)) {
+                        log_syntax(unit, LOG_ERR, filename, line, 0, "Invalid file descriptor name, ignoring: %s", name);
+                        return 0;
+                }
+                c->std_input = EXEC_INPUT_NAMED_FD;
+                return free_and_strdup(&c->std_input_fdname, name);
+        } else {
+                ExecInput ei = _EXEC_INPUT_INVALID;
+                config_parse_input(unit, filename, line, section,
+                                   section_line, lvalue, ltype,
+                                   rvalue, &ei, userdata);
+                if (ei != _EXEC_INPUT_INVALID)
+                        c->std_input = ei;
+                return 0;
+        }
+}
+
+int config_parse_exec_output(const char *unit,
+                             const char *filename,
+                             unsigned line,
+                             const char *section,
+                             unsigned section_line,
+                             const char *lvalue,
+                             int ltype,
+                             const char *rvalue,
+                             void *data,
+                             void *userdata) {
+        ExecContext *c = data;
+        ExecOutput eo = _EXEC_OUTPUT_INVALID;
+        const char *name;
+
+        assert(data);
+        assert(filename);
+        assert(line);
+        assert(lvalue);
+        assert(rvalue);
+
+        name = startswith(rvalue, "fd:");
+        if (name) {
+                /* Strip prefix and validate fd name */
+                if (!fdname_is_valid(name)) {
+                        log_syntax(unit, LOG_ERR, filename, line, 0, "Invalid file descriptor name, ignoring: %s", name);
+                        return 0;
+                }
+                eo = EXEC_OUTPUT_NAMED_FD;
+        } else {
+                config_parse_output(unit, filename, line, section,
+                                    section_line, lvalue, ltype,
+                                    rvalue, &eo, userdata);
+                if (eo == _EXEC_OUTPUT_INVALID)
```

Comment:
same as above, I think it would be nicer to avoid config_parse_output() here (and drop the macro invocation defining it) and just call exec_context_output_from_string() directly.